app/ts/background/background-startup.ts

```diff
@@ -116,6 +116,7 @@ export async function onContentScriptConnected(simulator: Simulator, port: brows
 }
 
 async function newBlockAttemptCallback(blockheader: EthereumBlockHeader, ethereumClientService: EthereumClientService, isNewBlock: boolean, simulator: Simulator) {
+	if (ethereumClientService.getChainId() !== simulator.ethereum.getChainId()) throw new Error(`Chain Id Mismatch, node is on ${ ethereumClientService.getChainId() } while simulator is on ${ simulator.ethereum.getChainId() }`)
```

Comment:
wouldn't be noticeable, but `getChainId` function returns could be stored in a variable to reduce calls
```suggestion
  const clientChainId = ethereumClientService.getChainId()
  const simulatorChainId = simulator.ethereum.getChainId()
	if (clientChainId !== simulatorChainId) throw new Error(`Chain Id Mismatch, node is on ${ clientChainId } while simulator is on ${ simulatorChainId }`)
```

Reply:
they are just functions that return a single variable right away so I don't think its worth to modify the oneliner to 3 lines